File: src/g1ipmlinkedlist/CS/IPMLinkedList.Algo/DisplayLinkedList.cs
// File: DisplayLinkedList.cs

namespace IPMLinkedList.Algo;

public static class DisplayLinkedList
{
    public static void DisplayWithForwardArrow(LinkedListNode? head)
    {
        LinkedListNode? temp = head;

        while (temp != null)
        {
            Write(temp.Data);
            temp = temp.Next;
            if (temp != null)
            {
                Write(" → ");
            }
            else
            {
                Write(" → null");
            }
        }
        WriteLine();
    }
}

File: src/g1ipmlinkedlist/Java/com/example/ipmlinkedlist/DisplayLinkedList.java
// File: DisplayLinkedList.java

package com.example.ipmlinkedlist;

public class DisplayLinkedList {
    public static void displayWithForwardArrow(LinkedListNode head) {
        LinkedListNode temp = head;
        while (temp != null) {
            System.out.print(temp.data);
            temp = temp.next;
            if (temp != null) {
                System.out.print(" → ");
            } else {
                System.out.print(" → null");
            }
        }
        System.out.println();
    }
}
